Dependencies/Authentication/Sources/Authentication/AuthenticationManager.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  AuthenticationManager.swift
+//
+//
+//  Created by Kamil Wójcicki on 03/10/2023.
+//
+
+import AuthenticationInterface
+import FirebaseAuth
+
+
+final class AuthenticationManager: AuthenticationManagerInterface {
+    
+    init() {
+        userStateListener()
+    }
+    
+    private func userStateListener() {
+        auth.addStateDidChangeListener { [weak self] auth, user in
+            guard let self = self else { return }
+            
+            if let user {
+                self.userID = user.uid
+                self.isAuthenticatedUser = true
+            }
+        }
+    }
+    
+    
+    var userID: String = ""
+    var isAuthenticatedUser: Bool = false
+    private var loginEventHandler: ((Bool) -> Void)?
+    private let auth = Auth.auth()
+    
+    
+    #warning("Not finished yet")
+//    private var userHandler: ((AuthenticationInterface.User?) -> Void)?
+//        lazy var userUpdates: AsyncStream<AuthenticationInterface.User?> = {
+//            AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+//                userHandler = { continuation.yield($0) }
+//            }
+//        }()
+    
+    private var user: AuthenticationInterface.User? {
+        didSet {
+            //userHandler?(user)
+            loginEventHandler?(user != nil)
+        }
+    }
+    
+    lazy var signInResult: AsyncStream<Bool> = {
+        AsyncStream(bufferingPolicy: .bufferingNewest(1)) { continuation in
+            loginEventHandler = { continuation.yield($0) }
+        }
+    }()
+    
+    func getAuthenticatedUser() throws -> Bool {
+        guard let auth = auth.currentUser else {
+            throw AuthErrorHandler.getAuthenticatedUserError
+        }
+        return true
+    }
```

Comment:
```suggestion  
     func getCurrentUser() throws -> User {
        try auth.currentUser ?? { throw AuthenticationError.unauthenticated }()
    }
```
Twoja funkcja nazywa się getAuthenticatedUser, ale zwraca Bool. Więc nazwa wprowadza w błąd gdyż po jej wykonaniu nie otrzymujesz Usera

Więc proponuje:
1. albo zmienić nazwę
2. albo func getAuthenticatedUser() throws -> User